app/models/event.rb

```diff
@@ -242,26 +250,30 @@ def calculate_progress
   end
 
   ##
-  # Returns end of the event
+  # Returns the room in which the event is scheduled
   #
-  def end_time
-    self.start_time + self.event_type.length.minutes
+  def scheduled_room
+    event_schedule.try(:room)
```

Comment:
Maybe you should add some validations to EventSchedule ;) Then you don't need the .try everywhere!